osu.Game/Beatmaps/FramedBeatmapClock.cs

```diff
@@ -121,6 +121,17 @@ protected override void LoadComplete()
         protected override void Update()
         {
             base.Update();
+
+            if (Source != null && Source is not IAdjustableClock && Source.CurrentTime < finalClockSource.CurrentTime)
```

Comment:
Can you assure me that this isn't going to have some weird unexpected effects on gameplay?

Reply:
Actually, just in general, I don't know how to reassure myself that this isn't going to cause some sort of catastrophic failure. It's called every frame even in song select when time is progressing normally. The track is practically always behind the final clock time isn't it?

Reply:
In all cases but `OsuGameBase`, the source is `IAdjustableClock` (ie. a track) which will never run this code.

Reply:
You raise a good point though, this should be checking `decoupledClock` not `finalClockSource` (as offsets should not be involved).

Of note, I tested before PRing this and nothing breaks even if this is called every frame.